consensus/polybft/blockchain_wrapper.go

```diff
@@ -152,20 +152,33 @@ func (p *blockchainWrapper) GetHeaderByHash(hash types.Hash) (*types.Header, boo
 
 // NewBlockBuilder is an implementation of blockchainBackend interface
 func (p *blockchainWrapper) NewBlockBuilder(
-	parent *types.Header, coinbase types.Address,
-	txPool txPoolInterface, blockTime time.Duration, logger hclog.Logger) (blockBuilder, error) {
+	parent *types.Header,
+	coinbase types.Address,
+	txPool txPoolInterface,
+	blockTime time.Duration,
+	logger hclog.Logger,
+) (blockBuilder, error) {
 	gasLimit, err := p.blockchain.CalculateGasLimit(parent.Number + 1)
 	if err != nil {
 		return nil, err
 	}
 
+	// Calculate next base fee based on the parent head
+	baseFee := p.blockchain.CalculateBaseFee(parent)
+
+	// Update base fee in tx pool so txs could be sorted properly.
+	// This is the right place to update a base fee because this function
+	// gets executed when creating a new block, i.e. in the beginning of
+	// a new block so there are no transactions yet.
+	txPool.SetBaseFee(baseFee)
```

Comment:
I would move `baseFee := p.blockchain.CalculateBaseFee(parent)` + `txPool.SetBaseFee(baseFee)` to `func (c *consensusRuntime) OnBlockInserted` right bellow `c.config.txPool.ResetWithHeaders(fullBlock.Block.Header)`. I realize that current solution will work but maybe its good to have everything in one place and to  update `baseFee` for all validators/nodes and not just proposer

Reply:
It went over the code and it seems like it would require some structural changes there. This is a bit out of the scope of this PR so I would preffer to address this comment within the next PR. Anyway, this is the good point.